### PR TITLE
Operations

### PR DESCRIPTION
**Pull request by 2013HORSEMEATSCANDAL**
## About the PR

When nukies spawn, they will receive an Operation name in the spawn message. "Operation Blue Spell is a go !" for example.

## Why / Balance

I think it's a fun cosmetic change that could lead to nukie rounds having more personality. People might remember Operation Cold Vermin fonder than they'd remember Nukie round 23453.

## Technical details

Simply adds a system that randomly picks words from two datasets, not unlike revenant names, to generate a random Operation Name in the nukie spawn message.

## Media

da79219f6f52a6831a3643eae1af4dd5

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
**Changelog**

:cl: 2013HORSEMEATSCANDAL
- add: Cybersun has begun naming their operations to make them easier to remember.

### CODE REVIEW

**Review by moonheart08 — APPROVED** [review]
thank you *checks notes\* 2013HORSEMEATSCANDAL for this neat little feature.

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 1]
```diff
@@ -793,7 +797,8 @@ private void OnMindAdded(EntityUid uid, NukeOperativeComponent component, MindAd
 
             if (nukeops.TargetStation != null && !string.IsNullOrEmpty(Name(nukeops.TargetStation.Value)))
             {
-                _chatManager.DispatchServerMessage(playerSession, Loc.GetString("nukeops-welcome", ("station", nukeops.TargetStation.Value)));
+                var operation_name = _randomMetadata.GetRandomFromSegments(new List<string> {"operationPrefix", "operationSuffix"}, " ");
```
Comment (moonheart08) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:800]: This seems like it'll cause the operation name to change for every new nukie? Might want to store it somewhere.

Comment (deltanedas) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:800]: store in the nukeops rule then just tell nukies the field

Comment (2013HORSEMEATSCANDAL) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:800]: oh yeah ill try to edit that, sorry it's the first time i add something

Comment (2013HORSEMEATSCANDAL) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:800]: would you recommend i just remove that line or... ?

**Comment by 2013HORSEMEATSCANDAL** [comment]
didnt change the fix for some reason

**Comment by deltanedas** [comment]
Operation Grey Tide :trollface:

**Comment by deltanedas** [comment]
use ProtoId<DatasetPrototype> pro

**Comment by 2013HORSEMEATSCANDAL** [comment]
> use ProtoId pro

what is protold pro ?

**Comment by deltanedas** [comment]
nvm i was fucked

**Comment by 2013HORSEMEATSCANDAL** [comment]
goddamnit it's messed up again

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 2]
```diff
@@ -344,12 +347,13 @@ private void OnRoundStart(EntityUid uid, NukeopsRuleComponent? component = null)
             return;
 
         component.TargetStation = _random.Pick(eligible);
+        component.OperationName = _randomMetadata.GetRandomFromSegments(new List<string> {"operationPrefix", "operationSuffix"}, " ");
```
Comment (EmoGarbage404) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:350]: add constant prototype validators at the top of the file.

these would look like:
```cs
    [ValidatePrototypeId<DatasetPrototype>]
    private const string OperationPrefixDataset = "operationPrefix";
```

Comment (ficcialfaint) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:350]: perhaps just
```cs
private const List<ProtoId<DatasetPrototype>> OperationNamesDatasets = new {"operationPrefix", "operationSuffix"};

Comment (DrSmugleaf) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:350]: > perhaps just
> 
> ```cs
> private const List<ProtoId<DatasetPrototype>> OperationNamesDatasets = new {"operationPrefix", "operationSuffix"};
> ```

You can't have a List const, any type used in a `const` must be a compile-time constant.
https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/constants

Comment (2013HORSEMEATSCANDAL) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:350]: then what do i do ?

Comment (EmoGarbage404) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:350]: just make 2 variables and make the list in the GetRandomFromSegments call

**Comment by Emisse** [comment]
This is fun I like this.

**Review by Emisse — APPROVED** [review]

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 3]
```diff
@@ -344,6 +362,7 @@ private void OnRoundStart(EntityUid uid, NukeopsRuleComponent? component = null)
...
```
Comment (ficcialfaint) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:365]:
```suggestion
        component.OperationName = _randomMetadata.GetRandomFromSegments(new List<string> {OperationPrefixDataset, OperationSuffixDataset}, " ");
```

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 4]
```diff
@@ -990,18 +1009,6 @@ private void SpawnOperativesForGhostRoles(EntityUid uid, NukeopsRuleComponent? c
...
-    /// </summary>
-    private void NotifyNukie(ICommonSession session, NukeOperativeComponent nukeop, NukeopsRuleComponent nukeopsRule)
-    {
-        if (nukeopsRule.TargetStation is not { } station)
-            return;
-
-        _chatManager.DispatchServerMessage(session, Loc.GetString("nukeops-welcome", ("station", station)));
-        _audio.PlayGlobal(nukeop.GreetSoundNotification, session);
-    }
-
```
Comment (ficcialfaint) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:1004]: Why did you move the method and removed the summary above it?

File: Resources/Locale/en-US/game-ticking/game-presets/preset-nukeops.ftl [hunk 5]
```diff
@@ -3,7 +3,7 @@ nukeops-description = Nuclear operatives have targeted the station. Try to keep
 
 nukeops-welcome =
     You are a nuclear operative. Your goal is to blow up {$station}, and ensure that it is nothing but a pile of rubble. Your bosses, the Syndicate, have provided you with the tools you'll need for the task.
-    Death to Nanotrasen!
+    Operation {$operation_name} is a go ! Death to Nanotrasen!
```
Comment (ficcialfaint) [inline Resources/Locale/en-US/game-ticking/game-presets/preset-nukeops.ftl:6]: `operation_name` looks long-ish, I'd just leave `name`. Don't forget to change it in the `Loc.GetString()` call
```suggestion
    Operation {$name} is a go! Death to Nanotrasen!
```

**Review by ficcialfaint — DISMISSED** [review]

File: Resources/Prototypes/Datasets/Names/Operation_prefix.yml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  - Large
+  - Exciting
+  - Tasty
+  - Violent
+  - Blunt
+  - Space
+  - War
+  - Simian
+  - Wild
+  - Enemy
```
Comment (metalgearsloth) [inline Resources/Prototypes/Datasets/Names/Operation_prefix.yml:105]: Please alphabetise.

File: Resources/Prototypes/Datasets/Names/Operation_suffix.yml [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  - Garden
+  - Pistol
+  - Rifle
+  - Cannon
+  - Legend
+  - Daze
+  - Pool
+  - Money
+  - Curse
+  - Bag
```
Comment (metalgearsloth) [inline Resources/Prototypes/Datasets/Names/Operation_suffix.yml:105]: Same here.

**Review by metalgearsloth — DISMISSED** [review]
